### PR TITLE
Followup on 149 and 151

### LINKED ISSUE #154: Light client errors follow-up
- [x] make sure we return Error everywhere and not Kind (https://github.com/interchainio/tendermint-rs/pull/149#discussion_r377873604)
 - [x] use anomaly::fail: https://github.com/interchainio/tendermint-rs/pull/151#discussion_r377891589
 - [x] not really error related but intro a `expires_at` variable https://github.com/interchainio/tendermint-rs/pull/149#discussion_r377874816

### PR DESCRIPTION
**Pull request by Liamsi**
closes #154 

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGES.md

### CODE REVIEW

File: tendermint/src/lite/verifier.rs [hunk 1]
```diff
@@ -15,30 +15,39 @@ use crate::lite::error::{Error, Kind};
...
-                    at: last_header.bft_time().into() + *trusting_period,
-                    now: *now,
-                }
-                .into());
-            }
-            Ok(())
+    let header_time: SystemTime = last_header.bft_time().into();
+    let expires_at = header_time.add(trusting_period);
+    // Ensure now > expires_at.
+    // TODO: shouldn't this be <= instead below:
```
Comment (Liamsi) [inline tendermint/src/lite/verifier.rs]: See discussion: https://github.com/interchainio/tendermint-rs/pull/149#discussion_r377874816

**Comment by Liamsi** [comment]
I don't understand why this seems to decrease test-coverage?

File: tendermint/src/lite/verifier.rs [hunk 2]
```diff
@@ -15,30 +15,39 @@ use crate::lite::error::{Error, Kind};
...
+use std::ops::Add;
 
 /// Returns an error if the header has expired according to the given
 /// trusting_period and current time. If so, the verifier must be reset subjectively.
 fn is_within_trust_period<H>(
     last_header: &H,
-    trusting_period: &Duration,
-    now: &SystemTime,
+    trusting_period: Duration,
+    now: SystemTime,
```
Comment (Liamsi) [inline tendermint/src/lite/verifier.rs]: Both `SytemTime` and `Duration` implement the `Copy` trait. Hence, these references where not needed.

File: tendermint/src/lite/error.rs [hunk 3]
```diff
@@ -17,8 +17,8 @@ pub enum Kind {
     Expired { at: SystemTime, now: SystemTime },
 
     /// Trusted header is from the future.
-    #[error("duration error {:?}", _0)]
-    DurationOutOfRange(#[from] SystemTimeError),
+    #[error("trusted header time is too far in the future")]
+    DurationOutOfRange,
```
Comment (Liamsi) [inline tendermint/src/lite/error.rs]: Maybe we should rename this error.

File: tendermint/src/lite/verifier.rs (hunk 1 above)
Comment (Liamsi) [reply tendermint/src/lite/verifier.rs]: The spec mentions `return header.Time + trustedPeriod > now` (for the non-error'ing case) and the go code also behaves accordingly. So this should indeed be `<=`!

Comment (Liamsi) [reply tendermint/src/lite/verifier.rs]: Changed accordingly in 301958fdbcc6763ec31390c99958a29984f81ebe

**Comment by Shivani912** [comment]
As discussed with @Liamsi I'm working on fixing issues with `voting_power_in` explained [here](https://github.com/interchainio/tendermint-rs/issues/145#issuecomment-583790278) and will be adding more rust-only tests

File: tendermint/src/lite/error.rs (hunk 3 above)
Comment (Liamsi) [reply tendermint/src/lite/error.rs]: https://github.com/interchainio/tendermint-rs/issues/137#issuecomment-585362689

**Comment by Shivani912** [comment]
Looks good to me! 👍

**Comment by Liamsi** [comment]
Thanks for the review @Shivani912 👍

**Pull request merged**